test/installer/scripts/omsadmin_systest.rb

```diff
@@ -94,15 +94,42 @@ def test_reonboard
 
   def test_reonboard_different_workspace_id
     do_onboard(TEST_WORKSPACE_ID, TEST_SHARED_KEY)
-    old_guid = get_GUID()
+    old_guid = get_GUID(@omsadmin_test_dir_ws1)
 
     output = do_onboard(TEST_WORKSPACE_ID_2, TEST_SHARED_KEY_2)
-    new_guid = get_GUID()
+    new_guid = get_GUID(@omsadmin_test_dir_ws2)
+
+    assert_not_equal(old_guid, new_guid, "The GUID should change when reonboarding with a different workspace ID")
+    assert_not_match(/Reusing/, output, "Should not be reusing GUID")
+  end
+
+  def test_onboard_two_workspaces
+    $log.info "Test results in #{@omsadmin_test_dir}"
+    output1 = do_onboard(TEST_WORKSPACE_ID, TEST_SHARED_KEY)
+    assert_match(/Generating certificate/, output1, "Did not find cert generation message for workspace 1")
+    post_onboard_validation(@omsadmin_test_dir_ws1)
+
+    output2 = do_onboard(TEST_WORKSPACE_ID_2, TEST_SHARED_KEY_2)
+    assert_match(/Generating certificate/, output2, "Did not find cert generation message for workspace 2")
+    post_onboard_validation(@omsadmin_test_dir_ws2)
+
+    output_list = `#{@omsadmin_script} -l`
+    assert_match(/#{TEST_WORKSPACE_ID}/, output_list, "Did not find workspace 1 in list")
+    assert_match(/#{TEST_WORKSPACE_ID_2}/, output_list, "Did not find workspace 2 in list")
+
+    remove_workspace(TEST_WORKSPACE_ID)
+    remove_workspace(TEST_WORKSPACE_ID, false)
+
+    output_list = `#{@omsadmin_script} -l`
+
+    assert_not_match(/#{TEST_WORKSPACE_ID}/, output_list, "Should not find workspace 1 in list")
+    assert_match(/#{TEST_WORKSPACE_ID_2}/, output_list, "Did not find workspace 2 in list")
+
+    output_removeall = `#{@omsadmin_script} -X`
+    assert_equal(0, $?.to_i, "Remove all workspace should succeed: #{output_removeall}")
 
-    # With current multihoming support, the GUID is re-used when onboarding to an additional workspace
-    # TODO Robbie change this when AGENT_GUID logic has been fully converted with multihoming
-    assert_equal(old_guid, new_guid, "The GUID should not change when reonboarding to another workspace ID")
-    assert_match(/Reusing/, output, "Should be reusing GUID")
+    output_list = `#{@omsadmin_script} -l`
+    assert_match(/No Workspace/, output_list, "Did not find workspace 2 in list")
```

Comment:
Change message in case of non-match: "Output list should contain no workspaces: #{output_list}"